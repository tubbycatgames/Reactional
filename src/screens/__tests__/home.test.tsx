import React from 'react'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import Home from '../home'

describe('Home', () => {
  it('should render', () => {
    const navigation = { navigate: () => {} }
    const render = renderer.create(<Home navigation={navigation} />)
    expect(render.toJSON()).toMatchSnapshot()
  })

  it('should bind a nav action on button press', () => {
    const navigate = jest.fn()
    const navigation = { navigate }
    const render = renderer.create(<Home navigation={navigation} />)

    const button = render.root.findByType(Button)
    button.props.onPress()
    expect(navigate).toHaveBeenCalledWith('Game')
  })
})
