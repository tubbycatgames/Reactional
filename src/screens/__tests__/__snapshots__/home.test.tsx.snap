// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#e1f3f8",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1f3f8",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#212121",
          },
          undefined,
        ]
      }
    >
      Reactional!
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#212121",
          },
          undefined,
        ]
      }
    >
      An accessible, open-source game about reactions
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#212121",
          },
          undefined,
        ]
      }
    >
      Meant for learning, guiding, and enjoyment
    </Text>
  </View>
</RCTSafeAreaView>
`;
